stdlib/src/utils/stringref.mojo

```diff
@@ -494,6 +494,8 @@ struct StringRef(
 
         return res
 
+    # TODO: remove this method later on when nothing depends on it anymore.
+    # It has already been copied to `StringSlice`.
     fn find(self, substr: StringRef, start: Int = 0) -> Int:
```

Comment:
**Suggestion** Instead of leaving this definition behind, could we go one step farther, and have this definition call into `StringSlice.find()`?

I think that would look something like:

```mojo
# Methods on struct StringRef
fn find(self, substr: StringRef, start: Int = 0) -> Int:
    var self_str = self.unsafe_as_string_slice()
    var substr_str = substr.unsafe_as_string_slice()

    return self_str.find(substr, start)
```

where `StringRef.unsafe_as_string_slice()` would be a new method with the ~definition:

```mojo
fn unsafe_as_string_slice[
    lifetime: AnyLifetime = ImmutableStaticLifetime
](self) -> StringSlice[lifetime]:
    """Returns a string slice to this StringRef data.
    
    Safety:
        * This `StringRef` must remain valid for the duration that the resulting `StringSlice` is used.
           This might require manual lifetime extension.
        * This `StringRef` must contain valid UTF-8 encoded data.
    """
    return StringSlice[lifetime](
        unsafe_from_utf8_strref=substr
    )
```

Reply:
I tried and it's not possible. It's called at compile time and the new find method in StringSlice is not compile-time friendly for unknown reasons.

Reply:
Makes sense, do you mind filing a GitHub issue for the follow-up refactoring so we don't forget to circle back here?

Reply:
Done: https://github.com/modularml/mojo/issues/3294